manageiq-loggers.gemspec

```diff
@@ -26,6 +26,7 @@ Gem::Specification.new do |spec|
 
   spec.add_development_dependency "bundler"
   spec.add_development_dependency "cloudwatchlogger"
+  spec.add_development_dependency "systemd-journal"
```

Comment:
This will cause journald to be pulled into dev machines, however I think there are checks for the gem's presence to modify the logger, so adding it may unintentional consequences.  @agrare ?  Am I right here or is it different since I last checked?

Reply:
Correct we want to let the app that includes this to define if the gem should be installed or not

Reply:
@Fryguy @agrare This shouldn't matter since it is a "development_dependency", at least from a `bundler` perspective, but I guess I am confused at what your definition of "dev machines" is:

- Developers of "this gem"
- Developers of ManageIQ/manageiq

Again, if it is the latter, only `activesupport` and `manageiq-password` is being pulled in for a `bin/update`.

Reply:
Right, a dev dependency shouldn't matter for deployments, but I think without it Travis will bomb.

Reply:
We didn't want developers of manageiq who are on macs (and don't have libsystemd) to try to use the `ManageIQ::Loggers::Journald` logger and have it bomb out.  Right now we use the existence of the `systemd-journal` gem being installed to indicate that it can be used.  We could install it but have some other method to check if it is able to be used or not.

Reply:
The presence of the gem will change the logger, so "dev machines", (i.e. Developers of ManageIQ/manageiq) will have this gem come in automatically thus changing their local logger.  This will likely fail on Macs.  We would need a change in ManageIQ itself to avoid this.

Reply:
> Even so, will the gem being in the local gem set be a problem? That is, someone devs this gem and it gets installed on their system...then they go over to manageiq and what happens?

I don't think that would matter, as bundler only updates the `$LOADPATH` with whatever it resolves from the `Gemfile` and `Gemfile.lock` (pretty sure).  For transparency, I was checking the `ManageIQ/manageiq` `Gemfile.lock` file for what dependencies it had pulled, and `cloudwatchlogger` was **not** one of them.

Again, I think this would be a problem for someone trying to `bundle` this project specifically on a Mac, so maybe a `unless ruby.platform.linux?` (or the "non-pseudo code equivalent") should be enough if that is the concern, but we could also update the `README.md` to say "you need this gem to be able to run the test suite" or something like that in place of this line, and update our travis.yml and/or `Gemfile` in this project accordingly.

Reply:
Alright, yes, it will bomb on a Mac with that change (thanks @jvlcek for testing), so I've removed it.

Reply:
Updated it so that it's added as a dev dependency only on linux, and the specs are filtered with a `:linux` tag. That means they'll be skipped on non-Linux platforms. I tested successfully on my Mac to verify.